activerecord/test/cases/arel/nodes/filter_test.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require_relative "../helper"
+
+module Arel
+  module Nodes
+    class ::FilterTest < Arel::Spec
+      describe "Filter" do
+        it "should add filter to expression" do
+          table = Arel::Table.new :users
+          _(table[:id].count.filter(table[:gdp_per_capita].gteq(40_000)).to_sql).must_be_like %{
+              COUNT("users"."id") FILTER (WHERE "users"."gdp_per_capita" >= 40000)
+            }
+        end
+
+        describe "as" do
+          it "should alias the expression" do
+            table = Arel::Table.new :users
+            _(table[:id].count.filter(table[:gdp_per_capita].gteq(40_000)).as("foo").to_sql).must_be_like %{
+              COUNT("users"."id") FILTER (WHERE "users"."gdp_per_capita" >= 40000) AS foo
+            }
```

Comment:
We can also use a more descriptive word than `foo` in this example:

```suggestion
            _(table[:id].count.filter(table[:income].gteq(40_000)).as("rich_users_count").to_sql).must_be_like %{
              COUNT("users"."id") FILTER (WHERE "users"."income" >= 40000) AS rich_users_count
            }
```